python/guide/performance/compression/float16.md

```diff
@@ -0,0 +1,243 @@
+<!--- "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY -->
+<!--- KIND, either express or implied.  See the License for the -->
+<!--- specific language governing permissions and limitations -->
+<!--- under the License. -->
+
+# Mixed precision training using float16
+
+In this tutorial we will walk through how one can train deep learning neural networks with mixed precision on supported hardware. We will first see how to use float16 (both with Gluon and Symbolic APIs) and then some techniques on achieving good performance and accuracy.
+
+## Background
+
+The computational resources required for training deep neural networks have been increasing lately, because of growing of the complexity and size of models. Mixed precision training allows us to reduce the utilization of the resources by using lower precision arithmetic. In this approach you can train using 16 bit floating points (half precision) while using 32 bit floating points (single precision) for output buffers of float16 computation. It allows to achieve the same accuracy as training with single precision, while decreasing the required memory and training or inference time. Training using that combination of single and half precision is called mixed precision training.
+
+The float16 data type is a 16 bit floating point representation according to the [IEEE 754 standard](https://ieeexplore.ieee.org/document/4610935). It has a dynamic range where the precision can go from 0.0000000596046 (highest, for values closest to 0) to 32 (lowest, for values in the range 32768-65536). Despite the inherent reduced precision when compared to single precision float (float32), using float16 has many advantages. The most obvious advantages are that you can reduce the size of the model by half allowing the training of larger models and using larger batch sizes. The reduced memory footprint also helps in reducing the pressure on memory bandwidth and lowering communication costs. On hardware with specialized support for float16 computation you can also greatly improve the speed of training and inference. The Volta range of Graphics Processing Units (GPUs) from Nvidia have [Tensor Cores](https://www.nvidia.com/en-us/data-center/tensorcore/) which perform efficient float16 computation. A tensor core allows accumulation of half precision products into single or half precision outputs. For the rest of this tutorial we assume that we are working with Nvidia's Tensor Cores on a Volta GPU.
+
+## Prerequisites
+
+- Volta range of Nvidia GPUs (e.g. AWS P3 instance)
+- Cuda 9 or higher
+- CUDNN v7 or higher
+
+This tutorial also assumes understanding of how to train a network with float32. Please refer to other tutorials [here](http://beta.mxnet.io/guide/index.html) to get started with Apache MXNet and Gluon API. This tutorial focuses on the changes needed to switch from float32 to mixed precision and tips on achieving the best performance with mixed precision.
```

Comment:
Better to include a specific tutorial on training, or crash course or something rather than the guide landing page.